src/main/kotlin/org/opensearch/replication/util/Coroutines.kt

```diff
@@ -91,6 +91,16 @@ suspend fun <Req: ActionRequest, Resp: ActionResponse>
     }
 }
 
+suspend fun <Req: ActionRequest, Resp: ActionResponse>
+        OpenSearchClient.suspendExecute(replicationMetadata: ReplicationMetadata?,
+                                           action: ActionType<Resp>, req: Req, injectSecurityContext: Boolean = false, defaultContext: Boolean = false) : Resp {
+    return if(replicationMetadata != null) {
```

Comment:
what is  the need for this ? Since we already have two versions of `suspendExecute`, a third version makes things more complex.

Reply:
Yes, trying to consolidate all the version of APIs now. After refactor, we can have single API for this